File: 02_PrimeiroFormulario/app/src/main/java/br/com/fiap/formulario/MainActivity.java
package br.com.fiap.formulario;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    EditText edtNome;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void enviar(View view){
        String nome = edtNome.getText().toString();
        if (nome.trim().isEmpty()){
            Toast.makeText(this, "Informe o nome corretamente " + nome, Toast.LENGTH_SHORT).show();
        }else{
            Toast.makeText(this, "Olá " + nome, Toast.LENGTH_SHORT).show();
        }
    }
}

File: Palindromos/app/src/main/java/br/com/fiap/palindromos/MainActivity.kt
package br.com.fiap.palindromos

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    lateinit var edtFrase : EditText

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        edtFrase = findViewById(R.id.edtFrase)
    }

    fun verificar(view : View){
        var txt = edtFrase.text.toString().trim().toLowerCase()

        if(txt.length < 3){
            Toast.makeText(this, "Informe corretamente uma frase", Toast.LENGTH_SHORT).show()
            return
        }

        txt = removerCarecteresEspeciais(txt).toString()
        val regex = "[^a-z]".toRegex()
        txt = regex.replace(txt, "")

        val frase = if (txt.equals(txt.reversed())){
            "É um Palíndromo"
        }else{
            "Não é um Palíndromo"
        }
        toast(frase)
    }

    fun toast(frase : String){
        Toast.makeText(this, frase, Toast.LENGTH_SHORT).show()
    }

    fun  removerCarecteresEspeciais(txt : String) {
        val txt = txt.replace("á", "a")
            .replace("â", "a")
            .replace("à", "a")
            .replace("ã", "a")
            .replace("é", "a")
            .replace("ê", "a")
            .replace("í", "i")
            .replace("ì", "i")
            .replace("ó", "o")
            .replace("ô", "o")
            .replace("ò", "o")
            .replace("õ", "o")
            .replace("", "o")
            .replace("ó", "o")
            .replace("ç", "c")
        return
    }
}